etl/src/state/retries.rs

```diff
@@ -0,0 +1,649 @@
+use chrono::{DateTime, Utc};
+use futures::StreamExt;
+use postgres::schema::TableId;
+use std::collections::{HashMap, HashSet};
+use std::sync::Arc;
+use std::time::Duration;
+use tokio::sync::{Mutex, Notify};
+use tokio::task::JoinHandle;
+use tokio::time::Instant;
+use tokio_util::time::{DelayQueue, delay_queue};
+use tracing::{debug, error, info};
+
+use crate::concurrency::signal::SignalTx;
+use crate::state::store::base::StateStore;
+use crate::workers::pool::TableSyncWorkerPool;
+
+/// Orchestrates retry mechanisms for failed table synchronization operations.
+///
+/// Manages two types of retries:
+/// - Manual retries that require user intervention
+/// - Timed retries that are automatically executed after a delay
+///
+/// The orchestrator uses a [`DelayQueue`] for managing timed retries and ensures
+/// that only the most recent retry attempt is kept for each table ID.
+#[derive(Debug, Clone)]
+pub struct RetriesOrchestrator<S> {
```

Comment:
Having a central orchestrator allows us to extend retry behaviors with precise tracking of retries per table, enabling timed retries with backoff mechanisms and upper limits on the number of executions.